test/e2e/specs/editor/blocks/list.spec.js

```diff
@@ -0,0 +1,878 @@
+		await page.keyboard.type( 'two' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.press( 'Enter' );
+
+		await expect.poll( editor.getEditedPostContent ).toBe(
+			`<!-- wp:list -->
+<ul><li>one</li><li></li><li>two</li></ul>
+<!-- /wp:list -->`
+		);
+
+		await page.keyboard.press( 'Enter' );
+
+		await expect.poll( editor.getEditedPostContent ).toBe(
+			`<!-- wp:list -->
+<ul><li>one</li></ul>
+<!-- /wp:list -->
+
+<!-- wp:paragraph -->
+<p></p>
+<!-- /wp:paragraph -->
+
+<!-- wp:list -->
+<ul><li>two</li></ul>
+<!-- /wp:list -->`
+		);
+
+		// Should remove paragraph without creating empty list item.
+		await page.keyboard.press( 'Backspace' );
+
+		// Should merge lists into one.
+		await page.keyboard.press( 'ArrowDown' );
+		await pageUtils.pressKeyTimes( 'ArrowLeft', 'two'.length );
+
+		await expect.poll( editor.getEditedPostContent ).toBe(
+			`<!-- wp:list -->
+<ul><li>one</li><li></li></ul>
+<!-- /wp:list -->
+
+<!-- wp:list -->
+<ul><li>two</li></ul>
+<!-- /wp:list -->`
+		);
+	} );
+
+	test( 'should split into two ordered lists with paragraph', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '1. one' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'two' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.press( 'Enter' );
+
+		await expect.poll( editor.getEditedPostContent ).toBe(
+			`<!-- wp:list {"ordered":true} -->
+<ol><li>one</li></ol>
+<!-- /wp:list -->
+
+<!-- wp:paragraph -->
+<p></p>
+<!-- /wp:paragraph -->
+
+<!-- wp:list {"ordered":true} -->
+<ol><li>two</li></ol>
+<!-- /wp:list -->`
+		);
+	} );
+
+	test( 'should split indented list item', async ( { editor, page } ) => {
+		await editor.insertBlock( { name: 'core/list' } );
+		await page.keyboard.type( 'one' );
+		await page.keyboard.press( 'Enter' );
+		await editor.clickBlockToolbarButton( 'Indent' );
+		await page.keyboard.type( 'two' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'three' );
+
+		await expect.poll( editor.getEditedPostContent ).toBe(
+			`<!-- wp:list -->
+<ul><li>one<ul><li>two</li><li>three</li></ul></li></ul>
+<!-- /wp:list -->`
+		);
+	} );
+
+	test( 'should be immeadiately saved on indentation', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await editor.insertBlock( { name: 'core/list' } );
+		await page.keyboard.type( 'one' );
+		await page.keyboard.press( 'Enter' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'm' );
+
+		await expect.poll( editor.getEditedPostContent ).toBe(
+			`<!-- wp:list -->
+<ul><li>one<ul><li></li></ul></li></ul>
+<!-- /wp:list -->`
+		);
+	} );
+
+	test( 'should change the base list type', async ( { editor, page } ) => {
+		await editor.insertBlock( { name: 'core/list' } );
+		const button = await page.waitForSelector(
+			'button[aria-label="Ordered"]'
+		);
+		await button.click();
```

Comment:
I think we should be able to change this to clicking on a role selector? Or maybe just `editor.clickBlockToolbarButton( 'Ordered' )`?

Reply:
Oh, 

Modified to use clickBlockToolbarButton.